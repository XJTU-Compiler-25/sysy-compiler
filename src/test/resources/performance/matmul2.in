version https://git-lfs.github.com/spec/v1
oid sha256:58aa89290034d5f302a0cc354cef6ffde3fad0462af29dae15fb5527b53c6edc
size 307946
