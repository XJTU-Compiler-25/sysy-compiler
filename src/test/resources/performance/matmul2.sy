int a[300][300];
int b[300][300];
int c[300][300];
const int MAX = 2147483647;

int main(){
    int n;
    int i;
    int j;
    int k;
    int sum = 0;

    i = 0;
    while(i<300)
    {
        n = getarray(a[i]);
        if(n!=300){
            return n;
        }
        i = i + 1;
    }

    starttime();

    i = 0;
    while(i<300){
        j = 0;
        while(j<300){
            b[i][j] = a[j][i];
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<300){
        j = 0;
        while(j<300){
            k = 0;
            int temp = 0;
            while(k<300){
                if(a[i][k]*b[k][j] % 2 == 0)
                    temp = temp + a[i][k]*b[k][j];
                k = k+1;
            }
            c[i][j] = temp;
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<300){
        j = 0;
        int temp = MAX;
        while(j<300){
            if(c[i][j]<temp)
            {
                temp = c[i][j];
            }
            j = j+1;
        }
        j = 0;
        while(j<300){
            c[i][j] = temp;
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<300){
        j = 0;
        int temp = MAX;
        while(j<300){
            c[i][j] = -c[j][i];
            j = j+1;
        }
        i = i+1;
    }

    i = 0;
    while(i<300){
        j = 0;
        int temp = MAX;
        while(j<300){
            sum = sum + c[i][j];
            j = j+1;
        }
        i = i+1;
    }

    stoptime();

    putint(sum);
    return 0;
}

