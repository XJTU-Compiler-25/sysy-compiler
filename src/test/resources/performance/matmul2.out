version https://git-lfs.github.com/spec/v1
oid sha256:5858bdf289b1b5c17186c0ea5bce009697f5557a684b90ac78855e0e32e4dedc
size 11
